src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -67,6 +68,26 @@ internal static partial class Ldap
     {
         static Ldap()
         {
+            // Register callback that tries to load other libraries when the default library "libldap-2.4.so.2" not found
+            AssemblyLoadContext.Default.ResolvingUnmanagedDll += (assembly, ldapName) =>
+            {
+                IntPtr handle = IntPtr.Zero;
+
+                if (ldapName != Libraries.OpenLdap)
+                {
+                    return handle;
```

Comment:
Not sure if we should throw instead in this case

Reply:
We shouldn't throw from this API. The system will provide a "missing DLL" exception for us.

Reply:
Thanks